tests/sysprep_test.go

```diff
@@ -0,0 +1,361 @@
+	gracePeriod := int64(0)
+	spinlocks := uint32(8191)
+	firmware := types.UID(windowsFirmware)
+	_false := false
+	return v1.VirtualMachineInstanceSpec{
+		TerminationGracePeriodSeconds: &gracePeriod,
+		Domain: v1.DomainSpec{
+			CPU: &v1.CPU{Cores: 2},
+			Features: &v1.Features{
+				ACPI: v1.FeatureState{},
+				APIC: &v1.FeatureAPIC{},
+				Hyperv: &v1.FeatureHyperv{
+					Relaxed:   &v1.FeatureState{},
+					VAPIC:     &v1.FeatureState{},
+					Spinlocks: &v1.FeatureSpinlocks{Retries: &spinlocks},
+				},
+			},
+			Clock: &v1.Clock{
+				ClockOffset: v1.ClockOffset{UTC: &v1.ClockOffsetUTC{}},
+				Timer: &v1.Timer{
+					HPET:   &v1.HPETTimer{Enabled: &_false},
+					PIT:    &v1.PITTimer{TickPolicy: v1.PITTickPolicyDelay},
+					RTC:    &v1.RTCTimer{TickPolicy: v1.RTCTickPolicyCatchup},
+					Hyperv: &v1.HypervTimer{},
+				},
+			},
+			Firmware: &v1.Firmware{UUID: firmware},
+			Resources: v1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceMemory: resource.MustParse("2048Mi"),
+				},
+			},
+			Devices: v1.Devices{
+				Disks: []v1.Disk{
+					{
+						Name:       windowsSealedDisk,
+						DiskDevice: v1.DiskDevice{Disk: &v1.DiskTarget{Bus: "sata"}},
+					},
+					{
+						Name:       "sysprep",
+						DiskDevice: v1.DiskDevice{CDRom: &v1.CDRomTarget{Bus: "sata"}},
+					},
+				},
+			},
+		},
+		Volumes: []v1.Volume{
+			{
+				Name: windowsSealedDisk,
+				VolumeSource: v1.VolumeSource{
+					Ephemeral: &v1.EphemeralVolumeSource{
+						PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: tests.DiskWindowsSysprep,
+						},
+					},
+				},
+			},
+			{
+				Name: "sysprep",
+				VolumeSource: v1.VolumeSource{
+					Sysprep: &v1.SysprepSource{
+						ConfigMap: &k8sv1.LocalObjectReference{
+							Name: "sysprepautounattend",
+						},
+					},
+				},
+			},
+		},
+	}
+
+}
+
+const (
+	windowsSysprepedVMIUser     = "Admin"
+	windowsSysprepedVMIPassword = "Gauranga"
+)
+
+var _ = Describe("[Serial][Sysprep]Syspreped VirtualMachineInstance", func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+
+	var windowsVMI *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+		tests.BeforeTestCleanup()
+		tests.SkipIfNoWindowsImage(virtClient)
+		tests.CreatePVC(tests.OSWindows, "35Gi", tests.Config.StorageClassWindows, true)
+		answerFileWithKey := insertProductKeyToAnswerFileTemplate(answerFileTemplate)
+		tests.CreateConfigMap("sysprepautounattend", map[string]string{"Autounattend.xml": answerFileWithKey, "Unattend.xml": answerFileWithKey})
+		windowsVMI = tests.NewRandomVMI()
+		windowsVMI.Spec = getWindowsSysprepVMISpec()
+		tests.AddExplicitPodNetworkInterface(windowsVMI)
+		windowsVMI.Spec.Domain.Devices.Interfaces[0].Model = "e1000"
+	})
+
+	Context("[ref_id:5105]should create the Admin user as specifide in the Autounattend.xml", func() {
```

Comment:
`specifide` -> `specified`

Reply:
I had to reread this 5 times to see the typo :P tnx!